util/util.go

```diff
@@ -703,6 +703,91 @@ func ManagerDelegatingClientFunc(cache cache.Cache, config *rest.Config, options
 	}, nil
 }
 
+// DelegatingClientFuncWithUncached returns a manager.NewClientFunc to be used when creating
+// a new controller runtime manager.
+//
+// A delegating client reads from the cache and writes directly to the server.
+// This avoids getting unstructured objects directly from the server.
+//
+// In addition, this function builds a custom reader that used the input list of objects
+// to determine which GVK should always query the live server, instead of going through a cache.
+func DelegatingClientFuncWithUncached(uncachedObjects ...runtime.Object) func(cache cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
```

Comment:
I found a little bit confusing having `ManagerDelegatingClientFunc` and `DelegatingClientFuncWithUncached` (without Manager prefix).
Have you consider to use a single method accepting a list of options? if options are nil, we fall back to the DelegatingClient with cache reader, otherwise, we use the new delegatingReader

Reply:
I dropped manager because these are/can be used outside of a manager. We'll also not need these functions in v1alpha4

Reply:
ok, thanks for the explanation